### PR TITLE
Adding a couple little unit tests for context.py

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

* Adding a couple little unit tests for context.py
* Also, I went through and cleaned a lot of rambling out of the `context.py` comments and docstrings.

## Why is the change being made?

GitHub will run all the unit tests in ARMI in series, and in parallel under MPI.  We also have GitHub run all the unit tests under Mac, Windows, and Linux.

So the tests here really aren't much, just a starting place. But they WILL be run in a full 2x3 grid of variations, which is useful for something as critical as `context.py`.

---

## Checklist

- [x] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [x] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify any new/changed code.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/developer/tooling.html#add-release-notes) have been updated if necessary.
- [x] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: armi/context.py [hunk 1]
```diff
@@ -93,33 +78,30 @@ def setMode(cls, mode):
 USER = getpass.getuser()
 START_TIME = time.ctime()
 
-# Set batch mode if not a TTY, which means you're on a cluster writing to a stdout file
-# In this mode you cannot respond to prompts or anything
-# This does not work replabliy for both windows and linux so an os-specific solution is
-# applied
+# Set batch mode if not a TTY, which means you're on a cluster writing to a stdout file. In this
+# mode you cannot respond to prompts. (This does not work replabliy for both Windows and Linux so an
```
Comment (opotowsky) [inline armi/context.py:82]:
```suggestion
# mode you cannot respond to prompts. (This does not work reliably for both Windows and Linux so an
```
catching my own typo!

File: armi/tests/test_context.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+from armi import context
+
+
+class TestContext(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def test_mpiSizeAndRank(self):
+        if context.MPI_SIZE > 1:
```
Comment (opotowsky) [inline armi/tests/test_context.py:26]: this would need to be added to the mpi unit tests to be hit, right?

File: armi/context.py [hunk 3]
```diff
@@ -15,10 +15,7 @@
 """
 Module containing global constants that reflect the executing context of ARMI.
 
-This contains information about the circumstatces under which an ARMI application is
-running. Things like the MPI environment, executing user, etc. live here. These are
-re-exported by the `armi` package, but live here so that import loops won't lead to
-as many issues.
+Things like the MPI environment, executing user, etc. live here.
```
Comment (jakehader) [inline armi/context.py:18]: Better name than `things`? Maybe "Implements global state attributes that are dependent on the environment, including serial and distributed memory parallelism, Operating System flags, modes of running the framework (e.g., interactive, batch), temporary storage of application files, etc."?

Comment (john-science) [reply armi/context.py]: Done!

**Pull request merged**